src/dsp_tools/commands/xmlupload/models/ingest.py

```diff
@@ -125,37 +117,31 @@ def _ingest(self, filepath: Path) -> IngestResponse:
             except requests.exceptions.RequestException as e:
                 raise PermanentConnectionError(f"{err}. {e}") from e
 
-    def get_bitstream_info(
-        self,
-        bitstream: XMLBitstream,
-        permissions_lookup: dict[str, Permissions],
-        res_label: str,
-        res_id: str,
-    ) -> tuple[bool, None | BitstreamInfo]:
-        """Uploads a file to the ingest server and returns the BitstreamInfo."""
+    def get_bitstream_info(self, file_info: IntermediaryFileValue) -> IngestResult:
+        """Uploads a file to the ingest server and returns the upload results."""
         try:
-            res = self._ingest(Path(self.imgdir) / Path(bitstream.value))
-            msg = f"Uploaded file '{bitstream.value}'"
+            res = self._ingest(Path(self.imgdir) / Path(file_info.value))
+            msg = f"Uploaded file '{file_info.value}'"
             logger.info(msg)
-            permissions = permissions_lookup.get(bitstream.permissions) if bitstream.permissions else None
-            return True, BitstreamInfo(bitstream.value, res.internal_filename, permissions)
+            return IngestResult(
+                True, BitstreamInfo(file_info.value, res.internal_filename, file_info.metadata.permissions)
+            )
         except PermanentConnectionError:
-            msg = f"Unable to upload file '{bitstream.value}' of resource '{res_label}' ({res_id})"
+            msg = f"Unable to upload file '{file_info.value}' of resource '{file_info.res_label}' ({file_info.res_id})"
             logger.opt(exception=True).warning(msg)
-            return False, None
+            return IngestResult(False, None)
 
 
 @dataclass(frozen=True)
 class BulkIngestedAssetClient(AssetClient):
     """Client for handling media info, if the assets were bulk ingested previously."""
 
-    def get_bitstream_info(
-        self,
-        bitstream: XMLBitstream,
-        permissions_lookup: dict[str, Permissions],
-        res_label: str,  # noqa: ARG002
-        res_id: str,  # noqa: ARG002
-    ) -> tuple[bool, BitstreamInfo | None]:
-        """Returns the BitstreamInfo of the already ingested file based on the `XMLBitstream.value`."""
-        permissions = permissions_lookup.get(bitstream.permissions) if bitstream.permissions else None
-        return True, BitstreamInfo(bitstream.value, bitstream.value, permissions)
+    def get_bitstream_info(self, file_info: IntermediaryFileValue) -> IngestResult:
+        """Returns the BitstreamInfo of the already ingested file based on the `IntermediaryFileValue.value`."""
+        return IngestResult(True, BitstreamInfo(file_info.value, file_info.value, file_info.metadata.permissions))
+
+
+@dataclass
+class IngestResult:
```

Comment:
The class IngestResult seems redundant. The success status doesn't need to be stored separately, it can be deduced from media_info: if it is None, then it wasn't successful.

This redundancy was not introduced by you, it was already present in the code. But I wouldn't put 2 redundant infos into a class.